mobile-ui/src/main/java/org/sagebionetworks/research/mobile_ui/perform_task/PerformTaskFragment.java

```diff
@@ -202,7 +216,7 @@ void showStep(StepView stepView) {
             SharedPreferences sharedPreferences = getContext().getSharedPreferences(sharedPreferencesKey,
                     Context.MODE_PRIVATE);
             sharedPreferences.edit().putLong(LAST_RUN_KEY, Instant.now().toEpochMilli()).apply();
-            // TODO handle end of perform task.
+            this.endPerformTask();
```

Comment:
Let's leave this as a TODO. I think if this fragment is to be removed, its removal should be done by its host.